packages/sync-service/lib/electric/schema.ex

```diff
@@ -0,0 +1,95 @@
+defmodule Electric.Schema do
+  import Bitwise
+
+  @bit_types ["bit", "varbit"]
+  @variable_length_character_types ["varchar", "text"]
+  @fixed_length_character_types ["bpchar"]
+  @time_types ["time", "timetz", "timestamp", "timestamptz", "interval", "date"]
+  @interval_field_masks [
+    %{unit: "YEAR", mask: 1 <<< 2},
+    %{unit: "MONTH", mask: 1 <<< 1},
+    %{unit: "DAY", mask: 1 <<< 3},
+    %{unit: "HOUR", mask: 1 <<< 10},
+    %{unit: "MINUTE", mask: 1 <<< 11},
+    %{unit: "SECOND", mask: 1 <<< 12}
+  ]
+  # -1 encoded as a 16 bit integer
+  @minus_1 65535
+  # all bits set in a 16 bit integer
+  @all_set 32767
```

Comment:
nitpick, to make it a little more self-documenting: 

```suggestion
  @all_set 0b0111111111111111
```